rts/System/Platform/Misc.cpp

```diff
@@ -519,4 +519,20 @@ std::string ExecuteProcess(const std::string& file, std::vector<std::string> arg
 	return execError;
 }
 
+void OpenURL(const std::string& url)
```

Comment:
I strongly dislike implementing non-trivial security checks myself, as it's easy to forget edge cases and those edge cases alone are enough to make the entire check pointless.
At the same time we'd lose some power, like being able to open directories, and I think these two operations are completely legitimate: 
`Spring.OpenURL(screenshotDir)`
`Spring.OpenURL(dataDir)`

Can/should we use a library to securely open URLs? 

Another alternative I'd like to consider is simply making games responsible for the things they do. If we restrict this to gadget unsynced, I think we can expect it to be used only by responsible people.